extractor/spotify/core.py

```diff
@@ -1,14 +1,31 @@
-from spotipy import Spotify
-from spotipy.oauth2 import SpotifyClientCredentials
-from dotenv import load_dotenv, find_dotenv
-import urllib.parse as parser
+from extractor.spotify.client import spotify_client
 
-load_dotenv(find_dotenv())
-# env variables: SPOTIPY_CLIENT_ID, SPOTIPY_CLIENT_SECRET
+class Track:
+    def __init__(self, id):
```

Comment:
Also, why are we initializing the class with an `id`? Does this come with the assumption that the caller knows the id in advance? Wouldn't that already involve calling the spotify API first?

Reply:
I did it that way so that the definition remains consistent with how the other classes are defined.  
There are two requests to the API: first when the `get_track` function is called. This sends a request to the `/search` endpoint. Here, we can modify our query in this function according to the information we already have from YT and maybe parse through multiple results from the request in the future. I thought that that logic should be independent from how we define the `Track` class.
 The second time, a request is made to the `/track` endpoint with the id we have from the first request. 
I thought this was a bit more systematic :sweat_smile: 
We can shift the search query to the `Track` class (like it was a commit back I think). Let me know you thoughts on this.

Reply:
Gotcha! I see the consistency. Its an extra network call for now, but lets optimize this later.